src/main/java/au/org/emii/ncdfgenerator/NcdfEncoderBuilder.java

```diff
@@ -0,0 +1,1476 @@
+			}
+		}
+	}
+
+
+	public NetcdfFileWriteable get() throws Exception
+	{
+		// TODO should just return a readable IStream, client shouldn't care that it's netcdf type.
+
+		try {
+			if( featureInstancesRS.next())
+			{
+				// munge
+				long instance_id = -1234;
+				Object o = featureInstancesRS.getObject(1);
+				Class clazz = o.getClass();
+				if( clazz.equals( Integer.class )) {
+					instance_id = (long)(Integer)o;
+				}
+				else if( clazz.equals( Long.class )) {
+					instance_id = (long)(Long)o;
+				} else {
+					throw new NcdfGeneratorException( "Can't convert intance_id type to integer" );
+				}
+
+				System.out.println( "whoot get(), instance_id is " + instance_id );
+
+				String selection = translate.process( selection_expr); // we ought to be caching the specific query ???
+
+				populateValues( definition.dimensions, definition.encoders,
+					"SELECT * FROM (" + definition.virtualInstanceTable + ") as instance where instance.id = " + Long.toString( instance_id) );
+
+
+				// is the order clause in sql part of projection or selection ?
+
+				// eg. concat "," $ map (\x -> x.getName) dimensions.values ...
+				String dimensionVar = "";
+				for( IDimension dimension : definition.dimensions.values() )
+				{
+					if( ! dimensionVar.equals("")){
+						dimensionVar += ",";
+					}
+					dimensionVar += "\"" + dimension.getName() + "\"" ;
+				}
+
+				populateValues( definition.dimensions, definition.encoders,
+					"SELECT * FROM (" + definition.virtualDataTable + ") as data where " + selection +  " and data.instance_id = " + Long.toString( instance_id) + " order by " + dimensionVar  );
+
+				NetcdfFileWriteable writer = createWritable.create();
+
+
+				for ( IDimension dimension: definition.dimensions.values()) {
+					dimension.define(writer);
+				}
+
+				for ( IVariableEncoder encoder: definition.encoders.values()) {
+					encoder.define( writer );
+				}
+				// finish netcdf definition
+				writer.create();
+
+				for ( IVariableEncoder encoder: definition.encoders.values()) {
+					// maybe change name writeValues
+					encoder.finish( writer );
+				}
+				// write the file
+				writer.close();
+
+				// TODO we should be returning a filestream here...
+				// the caller doesn't care that it's a netcdf
+				return writer;
+			}
+			else {
+				// no more netcdfs
+				conn.close();
+				return null;
+			}
+		} catch ( Exception e ) {
+			System.out.println( "Opps " + e.getMessage() );
+			conn.close();
+			return null;
+		}
+	}
+}
+
+
+public class NcdfEncoderBuilder
+{
+	// responsible for assembling the NcdfEncoder
+
+	public NcdfEncoderBuilder()
+	{
+		this.parser = new ExprParser();
+		this.translate = new  PGDialectTranslate();
+		this.createWritable = new CreateWritable( "testWrite.nc");
+	}
+
+	final IExprParser parser;
+	final IDialectTranslate translate;
+	final ICreateWritable createWritable;
+
+
+	public NcdfEncoder create ( InputStream config, String filterExpr, Connection conn ) throws Exception
```

Comment:
I know you haven't gotten to this yet, but from the web services viewpoint I don't think it needs to know about the config.  As discussed the web service knows about the typeName that's been requested, the cql filter that's been passed and can pass an output stream to which to write the zipped files.  Potentially the connection can be obtained from the client/container but not sure whether passing it is best here.